keyring/backend.py

```diff
@@ -45,8 +45,8 @@ class KeyringBackend(metaclass=KeyringBackendMeta):
     def __init__(self):
         self.set_properties_from_env()
 
-    # @abc.abstractproperty
```

Comment:
The intention here is to eventually make this method an abstractproperty so that a subclass would fails to construct if it doesn't define the property (failing faster).

It's my understanding that `NotImplementedError` is an older paradigm that preceded use of `abstractproperty`.

Wouldn't it be better to implement this as an abstract property? Does the typing system understand that an abstract property doesn't need an implementation?

Reply:
(eventually because it's liable to cause a backward incompatibility)

Reply:
But a abstractproperty is not a classproperty.

So if you want to have priority defined on class level there is no abstract concept for it to my knowledge in python.

Reply:
Ah. Good point and thanks for the reminder. In that case, then yes, I agree wit the change.